File: triplly-backend/src/entities/PlaceReview.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    CreateDateColumn,
    ManyToOne,
    JoinColumn,
    Unique,
    type Relation,
} from 'typeorm';
import type { Place } from './Place.js';
import type { User } from './User.js';

@Entity('place_reviews')
@Unique(['placeId', 'userId'])
export class PlaceReview {
    @PrimaryGeneratedColumn('uuid')
    id: string;

    @Column('uuid')
    placeId: string;

    @Column('uuid')
    userId: string;

    @Column({ type: 'int' })
    rating: number;

    @Column({ type: 'text' })
    content: string;

    @CreateDateColumn()
    createdAt: Date;

    @ManyToOne("Place", (place: Place) => place.reviews)
    @JoinColumn({ name: 'placeId' })
    place: Relation<Place>;

    @ManyToOne("User")
    @JoinColumn({ name: 'userId' })
    user: Relation<User>;
}

File: Triplly/Triplly/Models/PlaceReview.swift
import Foundation

struct PlaceReview: Codable, Identifiable, Equatable, Sendable {
    let id: String
    let placeId: String
    let userId: String
    let rating: Int
    let content: String
    let createdAt: String?
    let user: CheckInUser?
}

struct CreateReviewRequest: Codable, Sendable {
    let placeId: String
    let rating: Int
    let content: String
}
